File: backend/models/StudentProfile.js
const pool = require('../config/database');

class StudentProfile {
  static async createOrUpdate(userId, profileData) {
    const { full_name, phone, institution, class_level, medium, location, guardian_name } = profileData;
    console.log('StudentProfile.createOrUpdate called with userId:', userId, 'data:', profileData);
    const [result] = await pool.execute(
      `INSERT INTO student_profiles (user_id, full_name, phone, institution, class_level, medium, location, guardian_name)
       VALUES (?, ?, ?, ?, ?, ?, ?, ?)
       ON DUPLICATE KEY UPDATE
       full_name = VALUES(full_name),
       phone = VALUES(phone),
       institution = VALUES(institution),
       class_level = VALUES(class_level),
       medium = VALUES(medium),
       location = VALUES(location),
       guardian_name = VALUES(guardian_name)`,
      [userId, full_name, phone, institution, class_level, medium, location, guardian_name]
    );
    console.log('StudentProfile insert result:', result);
    return result;
  }

  static async findByUserId(userId) {
    const [rows] = await pool.execute('SELECT * FROM student_profiles WHERE user_id = ?', [userId]);
    return rows[0];
  }
}

module.exports = StudentProfile;

File: tutionmediaapp/lib/models/student_profile.dart
class StudentProfile {
  final int id;
  final int userId;
  final String? fullName;
  final String? phone;
  final String? institution;
  final String? classLevel;
  final String? medium;
  final String? location;
  final String? guardianName;

  StudentProfile({
    required this.id,
    required this.userId,
    this.fullName,
    this.phone,
    this.institution,
    this.classLevel,
    this.medium,
    this.location,
    this.guardianName,
  });

  factory StudentProfile.fromJson(Map<String, dynamic> json) {
    return StudentProfile(
      id: json['id'],
      userId: json['user_id'],
      fullName: json['full_name'],
      phone: json['phone'],
      institution: json['institution'],
      classLevel: json['class_level'],
      medium: json['medium'],
      location: json['location'],
      guardianName: json['guardian_name'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'full_name': fullName,
      'phone': phone,
      'institution': institution,
      'class_level': classLevel,
      'medium': medium,
      'location': location,
      'guardian_name': guardianName,
    };
  }
}